packages/parser/src/schemas/appsync.ts

```diff
@@ -0,0 +1,253 @@
+import { z } from 'zod';
+
+const AppSyncIamIdentity = z.object({
+  accountId: z.string(),
+  cognitoIdentityPoolId: z.string().nullable(),
+  cognitoIdentityId: z.string().nullable(),
+  sourceIp: z.array(z.string()),
+  username: z.string(),
+  userArn: z.string(),
+  cognitoIdentityAuthType: z.string().nullable(),
+  cognitoIdentityAuthProvider: z.string().nullable(),
```

Comment:
These fields are only present when you assume a role through a federation with Cognito Identity pools and then, right? If yes, make sense both be nullable.

Reply:
During my testing I made some GQL queries through the AppSync console using IAM authentication and printed out the `identity` field and all those fields were null.

Reply:
Thanks for confirming, this is what I expected and it makes sense make them `nullable`. When getting credentials from a Cognito Identity Pool, these values ​​should be
cognitoIdentityAuthType: authenticated/unauthenticated
cognitoIdentityAuthProvider: the name of the provider.